### ISSUE TITLE
Port should be optional

### ISSUE THREAD

**Issue by snotmare** [lbdremy/solr-node-client#219]
In our environment, we have an http host for solr so we don't need to provide the IP/port when connecting.  If I leave the port out of the config...
```
let solr = Solr.createClient({
	host: 'solrhost.local',
	path: '/solr',
	core: 'users',
	secure: 'true'
});
```
...solr will default to port 8983 in solr.js line 88...
`port : options.port || '8983',`

I would pass in an empty string, but the typings in index.d.ts require the port to be numeric...
```
export interface Options {
    host?: string;
    port?: number;
    core?: string;
    path?: string;
    agent?: string;
    secure?: string;
    bigint?: string;
    solrVersion?: string;
}
```

My only option right now is to comment out line 88 from solr.js. Can you take out the default for host and port? Or, can you provide another way to bypass the port?

Also, can you fix the typings so that the port can be numeric or a string to match up with what solr.js shows?

Thanks a bunch!

**Comment by nbaosullivan** [comment]
I ran into this as well, when using Solr as a service and have provided a pull request that offers a solution: set port to `0` if you want it ignored (when running into this I tried to specify 80 did not work as expected, but that was because I was using https *facepalm* so my actuall solution was just setting port to 443!).

**Comment by kibertoad** [comment]
Released in 0.8.0
